src/events.ts

```diff
@@ -41,10 +43,24 @@ export class EventEmitter {
 
     protected wrapListener(eventName: string, listener: (event: EventArgs) => void): (event: EventArgs) => void {
         return (event) => {
-            listener(new EventArgs(this, eventName, event));
+            const args = new EventArgs(this, eventName, event);
+            this.preProcessArgs(args);
+            const result = listener(args);
+            this.postProcessArgs(args);
+            return result;
         };
     }
 
+    protected preProcessArgs(args: EventArgs) {
+        // Do nothing in base
+    }
+
+    protected postProcessArgs(args: EventArgs) {
+        if (args && typeof args.returnValue !== "undefined") {
+            (<any>args.innerEvent).returnValue = args.returnValue;
```

Comment:
desktopJS events wrap the internal container events.  The returnValue on the djs event args needs to be propagated back to the inner event for the specific container.